### PR TITLE
Tests/teams unit tests

### PR DESCRIPTION
**Pull request by JoshuaHinman**
# Description

Unit tests for teams controller and services

## Issue link

Fixes # 86azu6kwr

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Feature updates / changes
- [x] Tests
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

yarn test:unit

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have updated the change log

### CODE REVIEW

File: src/teams/teams.controller.spec.ts [hunk 1]
```diff
@@ -1,25 +1,201 @@
...
+                repoUrlBE: expect.any(String),
+                deployedUrl: expect.any(String),
+                deployedUrlBE: expect.any(String),
+                tierId: expect.any(Number),
+                endDate: expect.any(Date),
+                createdAt: expect.any(Date),
+                updatedAt: expect.any(Date),
+            });
+
+            expect(mockTeamsService.findTeamById).toHaveBeenCalled();
```
Comment (cherylli) [inline src/teams/teams.controller.spec.ts:172]:
```suggestion
            expect(mockTeamsService.findTeamById).toHaveBeenCalledWith(1);
```

File: src/teams/teams.controller.spec.ts [hunk 2]
```diff
@@ -1,25 +1,201 @@
...
+                userId: expect.any(String),
+                voyageTeamId: expect.any(Number),
+                voyageRoleId: expect.any(Number),
+                statusId: expect.any(Number),
+                hrPerSprint: expect.any(Number),
+                createdAt: expect.any(Date),
+                updatedAt: expect.any(Date),
+            });
+
+            expect(mockTeamsService.updateTeamMemberById).toHaveBeenCalled();
```
Comment (cherylli) [inline src/teams/teams.controller.spec.ts:198]:
```suggestion
            expect(mockTeamsService.updateTeamMemberById).toHaveBeenCalledWith(
                1,
                requestMock,
                mockUpdateTeamMemberDto,
            );
```

**Review by cherylli — COMMENTED** [review]
looks good, all new and existing tests passed, I suggest updating toHaveBeenCalled to toHaveBeenCalledWith() in a couple of places

**Review by cherylli — APPROVED** [review]
Thanks. re-run the tests with the new changes, all passed

**Review by timDeHof — APPROVED** [review]
Looks good! All unit tests passed!

**Pull request merged**